srm.js

```diff
@@ -95,6 +95,29 @@ class SrmApiClient {
         return projectsResponse.data
     }
 
+    async getDefaultProjectBranchQueryParam(config) {
+        // If neither branches are specified, no need to add the query param
+        if (!config.baseBranchName && !config.targetBranchName)
+            return ``
+        // If both branches are specified, use the base branch as the default project branch. The target branch will
+        // later be used for analysis
+        else if (config.baseBranchName && config.targetBranchName)
+            return `"defaultBranchName": "${config.baseBranchName}"`
+        // For any other case, throw new project creation error
+        else throw Error(`New project creation failed because branch config is specified but incomplete. ` +
+            `Make sure to specify both 'base-branch-name' and 'target-branch-name', or neither, for project creation.`)
+    }
+
+    async createSrmProject(config) {
+        const requestBodyParams = [
+            `"name": "${config.projectName}"`,
+            await this.getDefaultProjectBranchQueryParam(config)
+        ]
+        const requestBody = JSON.parse(`{${requestBodyParams.filter(str => str).join()}}`)
```

Comment:
You can directly create + send JSON objects in JS:

```js
const projectData = { "name": config.projectName }

if (config.baseBranchName && config.targetBranchName)
   projectData.defaultBranchName = config.baseBranchName
else if (!!config.baseBranchName != !!config.targetBranchName)
  throw Error(...)

... this.http.post(..., projectData).catch(...)
```

The use of `!!` here forcefully converts the values to booleans, where "falsey" values (`null`, `undefined`, empty string, etc.) are converted to `false` and "truthy" values (an object, non-empty string, etc.) are converted to `true`

`if (x)` and `if (!x)` are a common way to check for whether a value was set, so comparing `!!x` and `!!y` is the same as comparing whether or not they were set:

```js
if (setBaseBranchName != setTargetBranchName) ...
```